modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2659,7 +2668,7 @@ def broadcast_apply_full_axis(
             apply_func=self._build_treereduce_func(axis, func),
             apply_indices=apply_indices,
             enumerate_partitions=enumerate_partitions,
-            keep_partitioning=True,
```

Comment:
The default value of `keep_partitioning`  for `apply_fill_axis` func must be `True` to keep the same behavior as before the change.